src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/killaura/features/KillAuraVelocityHit.kt

```diff
@@ -0,0 +1,111 @@
+package net.ccbluex.liquidbounce.features.module.modules.combat.killaura.features
+
+import com.google.common.collect.Lists
+import net.ccbluex.liquidbounce.config.types.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.events.PacketEvent
+import net.ccbluex.liquidbounce.event.events.TransferOrigin
+import net.ccbluex.liquidbounce.event.sequenceHandler
+import net.ccbluex.liquidbounce.event.tickHandler
+import net.ccbluex.liquidbounce.features.module.modules.combat.killaura.ModuleKillAura
+import net.ccbluex.liquidbounce.utils.client.Chronometer
+import net.ccbluex.liquidbounce.utils.client.PacketQueueManager
+import net.minecraft.network.packet.s2c.play.*
+
+object KillAuraVelocityHit : ToggleableConfigurable(ModuleKillAura, "VelocityHit", false) {
+
+    val extendRange by float("ExtendRange", 1.0f, 0.1f..2.0f)
+    private val whenLag by boolean("OnlyWhileLagging", false)
+
+    private var considerVelocityHit = false
+    private var damageReceived = false
+    private var onGroundTicks = 0
+    private var isPossible = false
+
+    private var timer = Chronometer()
+    private var lastPacketTime = Lists.newLinkedList<Long>()
```

Comment:
There are only a handful use cases for linked lists.

The `ArrayDeque` is the one that you are searching for. It is a ring buffer and has O(1) time complexity when inserting at/removing from the head *and* tail!

And documentation!